client/imports/modules/ui_components/index.js

```diff
@@ -167,6 +167,91 @@ UIComponents.prototype = {
       });
     },
 
+    setGridEditorValue({ selector, value }) {
+      // collect all keys
+      const allKeys = new Set();
+      value.forEach(row => {
+        Object.keys(row).forEach(k => allKeys.add(k));
+      });
+      // create HTML table
```

Comment:
I guess we need to provide `if (allKeys.size === 0) allKeys.add('Data');` just to prevent load issues if there's no data found